backend/src/database/state-migrations/migrations.ts

```diff
@@ -52,55 +54,59 @@ export const migrations: {
     8: treatmentSystemImprovements8,
 };
 
-export async function migrateInDatabaseTo(
-    targetStateVersion: number,
-    currentStateVersion: number,
+/**
+ *
+ * @param targetStateVersion The new state version the exercise should have.
+ *          `'latest'` to always migrate as far as possible.
+ *          Must be at least as big as the current sate version.
+ */
+export async function migrateInDatabase(
+    targetStateVersion: number | 'latest',
     exerciseId: UUID,
     entityManager: EntityManager
 ): Promise<void> {
-    const migrationFunctions = getMigrationFunctions(
-        currentStateVersion,
-        targetStateVersion
-    );
-    const stateMigrationFunctions = migrationFunctions.state;
-    const originalStates = await entityManager.findOne(ExerciseWrapperEntity, {
+    const exercise = await entityManager.findOne(ExerciseWrapperEntity, {
         where: { id: exerciseId },
-        select: { initialStateString: true, currentStateString: true },
     });
-    if (originalStates === null) {
+    if (exercise === null) {
         throw new RestoreError(
             'Cannot find exercise to convert in database',
             exerciseId
         );
     }
-    let initialState: object | null = null,
-        currentState: object | null = null;
-    if (stateMigrationFunctions.length > 0) {
-        initialState = JSON.parse(originalStates.initialStateString);
-        currentState = JSON.parse(originalStates.currentStateString);
-        stateMigrationFunctions.forEach((stateMigrationFunction) => {
-            initialState = stateMigrationFunction(initialState!);
-            currentState = stateMigrationFunction(currentState!);
-        });
-    }
-    const actionsMigrationFunctions = migrationFunctions.actions;
-    let actions: (object | null)[] | null = null;
-    if (actionsMigrationFunctions.length > 0) {
-        const originalActions = await entityManager.find(ActionWrapperEntity, {
+    const targetVersion =
+        typeof targetStateVersion === 'number'
+            ? targetStateVersion
+            : Math.max(...StrictObject.keys(migrations));
```

Comment:
Either 
* Change the `migrations` dictionary to an array where the index is the version and use `migrations.length`
* Assume (maybe with comment) that Object.keys().at(-1) is the highest stateNumber
* Make a public variable on the outer scope under `migrations` `const latestStateVersion = Math.max(...StrictObject.keys(migrations));`

Reply:
No, mich simpler: `ExerciseState.currentStateVersion` contains the current version. Its correctness is asserted by the test.